src/Ersatz/Relation/Prop.hs

```diff
@@ -145,3 +157,62 @@ transitive :: ( Ix a )
            => Relation a a -> Bit
 transitive r = implies (product r r) r
 
+-- | \( x \) is in the domain of /definition/ of a relation
+-- \( R \subseteq A \times B \) iff there is some \( y \in B \) such that
+-- \( (x, y) \in R \).
+in_domain_def :: ( Ix a, Ix b ) => Relation a b -> a -> Bit
+in_domain_def r a =
+    any ((r !) . (a,)) $ codomain r
+
+-- | \( y \) is in the image of a relation
+-- \( R \subseteq A \times B \) iff there is some \( x \in A \) such that
+-- \( (x, y) \in R \).
+in_image :: ( Ix a, Ix b ) => Relation a b -> b -> Bit
+in_image r b =
+    any ((r !) . (,b)) $ domain r
+
+-- | A relation \( R \subseteq A \times B \) is /left-total/ iff its domain of
+-- definition is identical to its domain; when \( R \) is a functional relation
+-- this means it defines a total rather than partial function.
+--
+-- For the property asserting that an /order/ relation is total ("strongly
+-- connected"), see 'total'.
```

Comment:
The namespace collision between "total [order] relation" and "total [functional] relation" is unfortunate. 

An alternative, heavierweight resolution might be to reorganize the internal modules of `Ersatz.Relation` [vertically, not horizontally](https://www.haskellforall.com/2021/05/module-organization-guidelines-for.html), with distinct modules for

 - Definitions that are principally about order relations.
 - Definitions that are principally about functional relations.

Reply:
An alternative fix:

Per the docstring of what is currently `total`, a synonym for the property it tests for is whether a relation is "*strongly* connected" and #78 added `connected` to `Relation.ARS`. 

`connected` is a property that any order relation could have (it isn't specific to abstract rewriting systems), so (regardless of where the definition lives internally) it could be re-exported by `Ersatz.Relation` and `total` could be renamed to `strongly_connected` for parallelism, freeing up `total`.